### PR TITLE
Android View + Activity Extensions

### PR DESCRIPTION
**Pull request by ReneeVandervelde**
Add some common Android extensions to Views and Activities

### CODE REVIEW

File: android/src/main/java/com/inkapplications/android/Views.kt [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+ *
+ *     fun toggleViews(show: Boolean) {
+ *         my_view.setVisibility(show)
+ *         my_invisible_view.setVisibility(show, View.INVISIBLE)
+ *     }
+ *
+ * @param visible Whether the view should be displayed or not.
+ * @param hiddenState Optional parameter to control how the view is hidden when [visible] is false.
+ */
+fun View.setVisibility(visible: Boolean, hiddenState: Int = View.GONE) {
```
Comment (AndrewReitz) [inline android/src/main/java/com/inkapplications/android/Views.kt:25]: Isn't this already in androidx?

**Review by AndrewReitz — APPROVED** [review]

File: android/src/main/java/com/inkapplications/android/Events.kt [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+ *    override fun onOptionsItemSelected(item: MenuItem): Boolean = when (item.itemId) {
+ *        android.R.id.home -> stopPropagation { onBackPressed() }
+ *        else -> continuePropagation()
+ *    }
+ * 
+ * @param action An optional action to take before stopping the event.
+ * @return Always returns true, to indicate that the event has been handled.
+ * @see continuePropagation for the opposite.
+ */
+inline fun stopPropagation(action: () -> Unit = {}): Boolean {
```
Comment (AndrewReitz) [inline android/src/main/java/com/inkapplications/android/Events.kt:24]: Is the default argument really necessary?

File: android/src/main/java/com/inkapplications/android/Views.kt (hunk 1 above)
Comment (ReneeVandervelde) [reply android/src/main/java/com/inkapplications/android/Views.kt:25]: [Citation Needed]?

File: android/src/main/java/com/inkapplications/android/Events.kt (hunk 2 above)
Comment (ReneeVandervelde) [reply android/src/main/java/com/inkapplications/android/Events.kt:24]: Idk, it just reads nicer if you're doing nothing: `return stopPropagation()` vs `return stopPropagation() {}`

File: android/src/main/java/com/inkapplications/android/Views.kt (hunk 1 above)
Comment (AndrewReitz) [reply android/src/main/java/com/inkapplications/android/Views.kt:25]: Ever since they moved it out of github I have no idea how to find anything in it :(

**Pull request merged**